lib/carrier/messaging/tracker.ex

```diff
@@ -0,0 +1,179 @@
+defmodule Carrier.Messaging.Tracker do
+
+  defstruct [reply_endpoints: %{}, subscriptions: %{}, monitors: %{}, unused_topics: []]
+
+  @spec add_reply_endpoint(Tracker.t, pid()) :: {Tracker.t, String.t}
+  def add_reply_endpoint(%__MODULE__{reply_endpoints: reps}=tracker, subscriber) do
+    case Map.get(reps, subscriber) do
+      nil ->
+        topic = make_reply_endpoint_topic()
+        reps = reps
+        |> Map.put(subscriber, topic)
+        |> Map.put(topic, subscriber)
```

Comment:
I get that it's internal; I was thinking more from a readability perspective than an API perspective.